### PR TITLE
[Gutenberg] Set consistent editor colours in both the WordPress and Jetpack apps

### PR DESCRIPTION
**Pull request by SiobhyB**
Fixes the Android side of https://github.com/wordpress-mobile/gutenberg-mobile/issues/5047, [iOS PR is here](https://github.com/wordpress-mobile/WordPress-iOS/pull/19113)

## Description

In the Jetpack app, some UI elements like the cursor, text selection, and caret are using a green colour. However, the rest of the editor’s elements are using blue. 

To be consistent with colours across the editor, this PR seeks to update them for consistency across both the WordPress and Jetpack apps. The PR specifically achieves the following for the Jetpack app:

* Updates the colour of UI elements within the main editor (e.g. cursor, highlight) from green to blue.
* Updates the accent of sub screens within Post Settings from green to blue.

There's still some known inconsistencies that need to be addressed in separate PRs, however:

* The media picker (available from media blocks and the Post Settings screen) still has some green.
* There is still green visible in other screens 'connected' to the editor, such as Preview, History, Help & Support, etc. 

## Screenshots

A before/after screenshot taken from the Jetpack app to show the editor's green → blue colour changes:

| Before  | After |
| ------------- | ------------- |
|  |  |

## Testing

* Download this PR's installable build for both the WordPress and Jetpack apps.
* In each app, navigate to the editor.
* Verify the colours used throughout the editor appear consistent for both app. The cursor, text selection, and caret should all now be blue for Jetpack (this is in contrast to the green that's used outside of the app's editor). There should be no change in the WordPress app, with the colours remaining blue.  

Special considerations to keep in mind while testing include:

* This PR aims to bring colour consistency to the Post Settings screen as well as the editor.
* Testing of any popovers, modals, or less visible UI elements within the editor would be beneficial. 
* As this is a UI change, dark/light mode should be a consideration when testing.
* These changes should be noticeable in both the post _and_ the page editor.
* There should be no changes to colours outside of the editor. Screens that may be used in more one place, such as category selection, should be tested.

### React Native RootView

* A new theme, `GutenbergContainerTheme` was [created under the editor's `styles.xml` file](https://github.com/wordpress-mobile/WordPress-Android/blob/91a25da6e13dfd06e7b6bf105fa00e13c2468d44/libs/editor/src/main/res/values/styles.xml#L7-L9). This theme references [an accent colour that equals the same hex code](https://github.com/wordpress-mobile/WordPress-Android/blob/91a25da6e13dfd06e7b6bf105fa00e13c2468d44/libs/editor/src/main/res/values/colors.xml#L33) as the WordPress app's primary colour. This new theme is then applied to [the editor's fragment.xml](https://github.com/wordpress-mobile/WordPress-Android/blob/91a25da6e13dfd06e7b6bf105fa00e13c2468d44/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml#L4).

### Post Settings 

* A new `colorPrimaryEditor` variable has been added to `colors_base.xml` for both WordPress ([light mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/main/res/values/colors_base.xml#L8) / [dark mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/main/res/values-night/colors_base.xml#L8)) and Jetpack ([light mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/jetpack/res/values/colors_base.xml#L8) / [dark mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/jetpack/res/values-night/colors_base.xml#L8)).
* This new variable has been used to create a special `PostSettingsTheme` in the main app's `style.xml` ([light mode](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/res/values-night/styles.xml#L129-L135) / [dark mode](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/res/values-night/styles.xml#L129-L135)). 
* `PostSettingsTheme` can now be applied to any areas in the Post Settings screen that should remain blue across both the WordPress and Jetpack apps. For example, the [input dialogs](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsInputDialogFragment.java#L87-L88) and [the add categories screen](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/res/layout/add_category.xml#L5).

## Regression Notes
1. Potential unintended areas of impact

There's the potential for this PR to unexpectedly change colours in either the WordPress or Jetpack app beyond what's desired. For example, the cursor colour may unexpectedly change within the WordPress app. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I manually tested both apps, making sure to experiment with various blocks and writing flows.

3. What automated tests I added (or what prevented me from doing so)

No automated tests.

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by SiobhyB** [comment]
@mkevins, I wondered if you might have availability to help review this PR? I definitely lack Android-specific experience and your expertise would be really valuable. 🙇‍♀️

**Review by dcalhoun — COMMENTED** [review]
This looks good to me. I tested both the WordPress and Jetpack apps on an Samsung Galaxy S20, both light and dark modes. I did not encounter any unexpected UI colors. 🎉 

I am interested to see @mkevins' feedback on the code before we merge. 

Thank you for your hard work to put this together!

**Comment by mkevins** [comment]
Hi @SiobhyB and @dcalhoun 👋 😄 

> @mkevins, I wondered if you might have availability to help review this PR? I definitely lack Android-specific experience and your expertise would be really valuable. 🙇‍♀️

I'm happy to have a look 👍 

I've taken the PR for a spin to test it out, and I found some strange behavior: the image and gallery blocks are invisible for some reason. I suspect that it isn't directly due to the changes in this PR, but somehow indirectly. For example, the images reappear when I revert this single line:

```patch
diff --git a/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml b/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
index 5ea25d451a..9777e8c01d 100644
--- a/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
+++ b/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:id="@+id/gutenberg_container"
-    android:theme="@style/GutenbergContainerTheme"
     android:layout_height="match_parent"
     android:layout_width="match_parent"
     android:background="@android:color/white"
```

I notice that it does not occur for other image-containing blocks, such as media-text, cover, etc. so I suspect it has something to do with an issue with FastImage. I found a similar issue that has been addressed with a work-around: https://github.com/WordPress/gutenberg/issues/42869 and sure enough, invoking that work-around in this PR (i.e. rotating the device to trigger the fallback `RNImage` implementation to be used instead) causes the images to re-appear.

I haven't gotten to the root of why this is happening, but I've tried a few things to investigate it. Since we are indeed using a custom Glide module, I added [this](https://github.com/DylanVann/react-native-fast-image/blob/main/docs/app-glide-module.md):

```patch
diff --git a/packages/react-native-bridge/android/build.gradle b/packages/react-native-bridge/android/build.gradle
index 99069d55cf..064b8eb015 100644
--- a/packages/react-native-bridge/android/build.gradle
+++ b/packages/react-native-bridge/android/build.gradle
@@ -1,6 +1,7 @@
 buildscript {
     ext {
         willPublishReactNativeBridgeBinary = properties["willPublishReactNativeBridgeBinary"]?.toBoolean() ?: false
+        excludeAppGlideModule = true
     }
 }
 
diff --git a/packages/react-native-editor/android/build.gradle b/packages/react-native-editor/android/build.gradle
index 848c570734..05bb0c8a33 100644
--- a/packages/react-native-editor/android/build.gradle
+++ b/packages/react-native-editor/android/build.gradle
@@ -8,6 +8,7 @@ buildscript {
         targetSdkVersion = 30
         supportLibVersion = '28.0.0'
         ndkVersion = "21.4.7075529"
+        excludeAppGlideModule = true
     }
     repositories {
         google()
```

But this did not resolve the issue. Disabling FastImage _does_ resolve the issue, so perhaps that is an option until we uncover the root of the issue (cc: @geriux in case you have any more insight about that, or know whether disabling FastImage is feasible).

To explore more about the issue, I also tried using the parent theme directly (i.e. bypassing the other changes from this PR):

```patch
diff --git a/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml b/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
index 5ea25d451a..3f251531f2 100644
--- a/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
+++ b/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:id="@+id/gutenberg_container"
-    android:theme="@style/GutenbergContainerTheme"
+    android:theme="@style/Theme.MaterialComponents.Light.DarkActionBar.Bridge"
     android:layout_height="match_parent"
     android:layout_width="match_parent"
     android:background="@android:color/white"
```

and even applying just this single change directly on `trunk`:

```patch
diff --git a/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml b/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
index 9777e8c01d..3f251531f2 100644
--- a/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
+++ b/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:id="@+id/gutenberg_container"
+    android:theme="@style/Theme.MaterialComponents.Light.DarkActionBar.Bridge"
     android:layout_height="match_parent"
     android:layout_width="match_parent"
     android:background="@android:color/white"
``` 

and this was enough to cause the invisible images with FastImage.

I will investigate this further.

**Comment by geriux** [comment]
> (cc: @geriux in case you have any more insight about that, or know whether disabling FastImage is feasible).

@mkevins I think we should while we work on fixing this issue on Android. I'll create a GitHub issue so we can track this. Thanks for the ping and for testing different scenarios 🙇

**Comment by mkevins** [comment]
Circling back to mention that I investigated the disappearing images further [here](https://github.com/WordPress/gutenberg/issues/43149#issuecomment-1213440827), but did not discover the root cause of that issue.

Aside from the unexpected issue described above (which seems to only be tangentially related with the changes in this PR), the approach in this PR looks correct to me. Nice work making these changes. When the FastImage issue is resolved (either temporarily by disabling it, or by resolving the root cause), I think it'd be good to merge `trunk` into this PR and test one more time to be sure and it should be good to merge, imo. @geriux 👋 😄 do you think we can land a PR to disable FastImage until we fix this? I'm happy to open a PR or review one to do that if you think this is the best way forward for now. Wdyt?

**Comment by geriux** [comment]
> . @geriux 👋 😄 do you think we can land a PR to disable FastImage until we fix this?

Yes I think we should, I'll open a PR to do that. I'll take that offer to review it 😄 Thanks!

**Comment by geriux** [comment]
I've created the PR to [disable](https://github.com/WordPress/gutenberg/pull/43322) FastImage on Android.

**Comment by dcalhoun** [comment]
Now that FastImage is disabled on Android in `gutenberg-mobile`, we need to integrate a `gutenberg-mobile` alpha release into WPAndroid to prep this PR for review again. I plan to wait until after the 20.6 release cut (Monday) to do so and move this PR forward, as we do not want to increment the `gutenberg-mobile` version for 20.6.

**Comment by fluiddot** [comment]
> Now that FastImage is disabled on Android in `gutenberg-mobile`, we need to integrate a `gutenberg-mobile` alpha release into WPAndroid to prep this PR for review again. I plan to wait until after the 20.6 release cut (Monday) to do so and move this PR forward, as we do not want to increment the `gutenberg-mobile` version for 20.6.

@dcalhoun I agree on not merging this PR until version 20.6 is cut, due to the need of integrating the disabling of FastImage into WPAndroid. However, I think that for moving forward the PR review and testing, maybe we could just update the Gutenberg Mobile reference to point to the latest commit of [the Gutenberg Mobile repo](https://github.com/wordpress-mobile/gutenberg-mobile), wdyt?

**Comment by dcalhoun** [comment]
> [...] I agree on not merging this PR until version 20.6 is cut [...] However, I think that for moving forward the PR review and testing, maybe we could just update the Gutenberg Mobile reference to point to the latest commit of [the Gutenberg Mobile repo](https://github.com/wordpress-mobile/gutenberg-mobile), wdyt?

Indeed, a great point. I will do that.

File: build.gradle [hunk 1]
```diff
@@ -10,7 +10,7 @@ plugins {
 ext {
     wordPressUtilsVersion = '2.7.0'
     wordPressLoginVersion = 'trunk-657bed28fa735780e3ffcc214503a9e5d1286d6c'
-    gutenbergMobileVersion = 'v1.81.0'
+    gutenbergMobileVersion = 'trunk-c39088daab5b49b48db81b01bc1e143999d8396b'
```
Comment (dcalhoun) [inline build.gradle]: Friendly reminder: we'll need to change this to a `gutenberg-mobile` alpha release prior to merging this PR.

**Review by dcalhoun — COMMENTED** [review]
I retested the color changes in the Jetpack and WordPress mobile apps on a Samsung Galaxy S20. 👍🏻 

@mkevins I did not encounter any of the [aforementioned issues](https://github.com/wordpress-mobile/WordPress-Android/pull/16968#issuecomment-1209658635) with the _Image_ or _Gallery_ block. However, I was unable to reproduce this issue with the older build either. Would you be willing to verify the invisible images are no longer an issue for you? 

Once we are able to integrate a `gutenberg-mobile` alpha release after the 20.5 release cut, I believe can move forward merging this PR.

**Comment by mkevins** [comment]
> However, I was unable to reproduce this issue with the older build either.

Oh interesting! Would you mind adding the testing details (phone, API version, etc.) for the scenario in which you could not reproduce the invisible images? I think this could be really helpful in narrowing down the root cause of the FastImage issue (since I believe there is still an effort underway to re-enable that once the root of the issue is found). Perhaps adding a note to [this issue](https://github.com/WordPress/gutenberg/issues/43149#issuecomment-1213440827) would be the best place for it, if you have a moment.

> Would you be willing to verify the invisible images are no longer an issue for you?

Sure thing! I'll test this today. 👍 Thanks for pushing the ref update!

**Comment by dcalhoun** [comment]
> Would you mind adding the testing details (phone, API version, etc.) for the scenario in which you could not reproduce the invisible images?

I tested [`pr16968-2b98958`](https://d2twmm2nzpx3bg.cloudfront.net/ea0ffdd4a98eeacfb5a47250241399da8490ef4e/wordpress-installable-build-pr16968-2b98958.apk) with a Samsung Galaxy S20 (Android 12). I added both _Image_ and _Gallery_ blocks a few times, and rotated the device orientation. I will share this on https://github.com/WordPress/gutenberg/issues/43149 as well. 

> Sure thing! I'll test this today. 👍 Thanks for pushing the ref update!

Thank you!

**Review by mkevins — APPROVED** [review]
These changes look good, and I have tested this with the latest gutenberg-mobile reference update (which brings changes to disable FastImage), and this is working as expected. I tested all the flows mentioned in the description, as well as exploring a few menus / settings (i.e. URL input for image links, etc.), and the colors are working as described. Nice work!

**Pull request merged**